### ISSUE TITLE
"empty" values of newImage differ from documented

### ISSUE THREAD

**Issue by mcclure** [bjornbytes/lovr#648]
Consider this simple program:

```
local material
local sampler

function lovr.load()
  local tdSize = 10
  local tdLim = tdSize-1
  local td = lovr.data.newImage(tdSize,tdSize)
  for x=0,tdLim do for y=0,tdLim do -- This admittedly dense code generates a checkerboard icon
    if x==0 or y==0 or x==tdLim or y==tdLim or (x+y)%2==0 then
      td:setPixel(x,y,1,1,1,1)
    end
  end end
  local texture = lovr.graphics.newTexture(td)
  sampler = lovr.graphics.newSampler{filter="nearest", wrap="clamp"}
  material = lovr.graphics.newMaterial{texture=texture}
end

function lovr.draw(pass)
  pass:setMaterial()
  pass:setSampler()
  pass:setColor(1,0,0,1)
  pass:plane(0,0,-5, 2,2)

  pass:setMaterial(material)
  pass:setSampler(sampler)
  pass:setColor(1,1,1,1)
  pass:plane(0,0,-3, 1,1)
end
```

This creates a texture in which the border and also every other pixel has been filled white, and all other pixels have been ignored.

Per the documentation for `newImage`: https://lovr.org/docs/lovr.data.newImage

```
format,	TextureFormat,	rgba	The format of the texture's pixels.
data,	Blob,	nil,	Raw pixel values to use as the contents. If nil, the data will all be zero.
```

There are two problems here. One, `rgba` seems to be a typo for `rgba8`.

But two, although I pass nil for the data argument above, the values are *not* initialized to zero. When I run this code, I get:

![image](https://user-images.githubusercontent.com/277318/222335748-df234cf5-c27a-4cbf-a983-6b8ee0de3eb0.png)

The uninitialized values seem to be set to 0xFFFFFFAA or something. I can confirm this by amending the checkerboard routine in the sample code with:

```
    else
      td:setPixel(x,y,0,0,0,0)
```

or:

```
    else
      td:setPixel(x,y,0,0,0,1)
```

…and it looks different.

It **wasn't** like this in 0.15. I ran this code there previously and it worked the way the current documentation describes.

**Comment by mcclure** [comment]
Note, this is dev 886db5f (+ trivial patches), I haven't had a chance to test on master

**Comment by mcclure** [comment]
By the way, I think MSVC's compiler when building for debug initializes all uninitialized memory to 0xCC, so what I'm seeing here might just be a side-effect of the buffer being allocated but not filled with anything at all.

**Comment by bjornbytes** [comment]
This was an unintended change in v0.16.0 and has been fixed in 6d382c2f0c6dec8c3064af96f2c30f9fe300e5a5 on master/dev.
